engine/common/provider/engine.go

```diff
@@ -133,9 +133,6 @@ func (e *Engine) process(originID flow.Identifier, message interface{}) error {
 	e.metrics.MessageReceived(e.channel.String(), metrics.MessageEntityRequest)
 	defer e.metrics.MessageHandled(e.channel.String(), metrics.MessageEntityRequest)
 
-	e.unit.Lock()
```

Comment:
this lock is not protecting anything, as the retrieve method is concurrent safe.

The provider engine is used in:
- collection node [providing collections](https://github.com/onflow/flow-go/blob/v0.28/cmd/collection/main.go#L381)
- execution node providing its own receipts](https://github.com/onflow/flow-go/blob/v0.28/cmd/execution_builder.go#L885)

Both retrieve functions are already concurrent safe, so we can safely remove this lock